src/model/import.ts

```diff
@@ -27,7 +27,7 @@ import {Severity, Warning} from './warning';
  * @template N The AST node type
  */
 export class ScannedImport implements Resolvable {
-  type: 'html-import'|'html-script'|'html-style'|'js-import'|string;
+  readonly type: 'html-import'|'html-script'|'html-style'|'js-import'|string;
```

Comment:
`keyof ImportFeatureKindMap` would be useful here. `FeatureKindMap` can extend it, and we'd reduce some duplication.